src/pages/home/report/ReportFooter.tsx

```diff
@@ -90,7 +93,10 @@ function ReportFooter({
     const isAnonymousUser = session?.authTokenType === CONST.AUTH_TOKEN_TYPES.ANONYMOUS;
 
     const isSmallSizeLayout = windowWidth - (isSmallScreenWidth ? 0 : variables.sideBarWidth) < variables.anonymousReportFooterBreakpoint;
-    const hideComposer = !ReportUtils.canUserPerformWriteAction(report, reportNameValuePairs) || blockedFromChat;
+
+    // If a user just signed in and is viewing a public report, optimistically show the composer while loading the report, since they will have write access when the response comes back.
+    const showComposerOptimistically = ReportUtils.isPublicRoom(report) && reportMetadata?.isLoadingInitialReportActions;
```

Comment:
Yep. `OpenReport` is taking an awfully long time. Anyway, I notice an issue where the composer will get shown optimistically even for an anonymous user which does not seem correct as demonstrated below. Here, we can additionally check for `isAnonymousUser` to fix this as suggested below. Does this look fine?

```suggestion
    const showComposerOptimistically = ReportUtils.isPublicRoom(report) && reportMetadata?.isLoadingInitialReportActions && !isAnonymousUser;
```

Along with the suggested change, code LGTM and tests well. I have added the test videos in the checklist. 


https://github.com/Expensify/App/assets/3004659/20f88365-a5b1-47b5-b47f-6ee0f887a756

Reply:
Great catch and solution. Thank you. I made the update.